### PR TITLE
fix: fix silent failure to add data of more than ~16MB via add_bytes or add_bytes_named

### PR DESCRIPTION
**Pull request by rklaehn**
## Description

- Make sure an error in send_all is not silently ignored but at least communicated to the remote side.
- Split large Bytes in add_bytes into chunks so we don't exceed MAX_FRAME_SIZE of 1024 * 1024 * 16
- Add some tests that use quinn

## Breaking Changes

None

## Change checklist

- [ ] Self-review.
- [ ] Documentation updates following the [style guide](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text), if relevant.
- [ ] Tests if relevant.
- [ ] All breaking changes documented.

### CODE REVIEW

File: Cargo.lock [hunk 1]
```diff
@@ -363,6 +372,15 @@ version = "0.1.2"
 source = "registry+https://github.com/rust-lang/crates.io-index"
 checksum = "597bb81c80a54b6a4381b23faba8d7774b144c94cbd1d6fe3f1329bd776554ab"
 
+[[package]]
+name = "bincode"
```
Comment (dignifiedquire) [inline Cargo.lock:376]: why is this making a comback?

Comment (rklaehn) [reply Cargo.lock:376]: what? How did this get here?

Comment (rklaehn) [reply Cargo.lock:376]: Ok, it was via quic-rpc, where I forgot to remove the feature from tokio-serde.

It is gone now:
```
❯ cargo tree --all-features -i bincode
error: package ID specification `bincode` did not match any packages
```

Comment (matheus23) [reply Cargo.lock:376]: But the `Cargo.lock` file here still has `bincode` here, right?
Why does `cargo tree` tell you it's not in there, but the `Cargo.lock` file in this PR still adds it?

Comment (dignifiedquire) [reply Cargo.lock:376]: @rklaehn did you give up fixing this?

Comment (rklaehn) [reply Cargo.lock:376]: will be in the next PR.

**Pull request merged**